sympy/sets/conditionset.py

```diff
@@ -190,11 +195,33 @@ def ok_sig(a, b):
                 ok_sig(i, j) for i, j in zip(a, b))
         if not ok_sig(self.sym, other):
             return S.false
+        # if either argument of And is False,
+        # we must return False even if we
+        # get a TypeError on the other
+
+        # try doing base_cond first and return
+        # False immediately if it is False
         try:
-            return And(
-                Contains(other, self.base_set),
-                Lambda((self.sym,), self.condition)(other))
+            base_cond = Contains(other, self.base_set)
+            base_error = False
         except TypeError:
```

Comment:
Where does the `TypeError` come from?

(I realise that you didn't add that but I'm just wondering since I look at this)

Reply:
The only time I experienced the `TypeError` was when it creates the lambda function and then evaluates it. If the lambda function expects something similar to symbols, and you pass in an (unevaluated) function (like `f` in the tests), it would raise.

I couldn't raise a `TypeError` with `base_cond` but I'd rather be safe than accidentally break a user's code.

Reply:
Fair enough. I just asked because I would like to remove most cases of `except TypeError` from the codebase as it can easily hide bugs.

Reply:
Actually I think it's fine to have an exception when using an unevaluated function `f` since `f` is not an instance of `Basic`.

Reply:
I would prefer it as it is because, functions are mathematical objects in their own right. So we could have functions being elements of sets and also not being elements of sets. 

I think other parts of the code base also assume it will not throw in cases such as this. That's kind of how I came across it. This method would sometimes return an unevaluated expression while working in `solveset`. Thus a `TypeError` must have been raised. But `And(False, <TypeError>)` should be `False` and that's why I decided to change it.

I can remove the try-catch for `base_cond` because I couldn't find out how it would raise. But I didn't want to cause problems with users' code where not having it might cause problems.